### PR TITLE
lcms, link against libpng16

### PR DESCRIPTION
**Pull request by Begasus**

### CODE REVIEW

File: media-libs/lcms/lcms-2.9.recipe [hunk 1]
```diff
@@ -23,14 +23,17 @@ if [ "$targetArchitecture" = x86_gcc2 ]; then
...
-	cmd:psicc$commandSuffix
-	cmd:tificc$commandSuffix
-	cmd:transicc$commandSuffix
-	lib:liblcms2$secondaryArchSuffix = 2.0.8 compat >= 2
+	lcms$secondaryArchSuffix = $portVersion
+	cmd:jpgicc$commandSuffix = 3.2
+	cmd:linkicc$commandSuffix = 2.2
+	cmd:psicc$commandSuffix = 2.1
+	cmd:tificc$commandSuffix = 6.2
+	cmd:transicc$commandSuffix = 4.3
```
Comment (Begasus) [inline media-libs/lcms/lcms-2.9.recipe]: Shouldn't the cmd's go to a _tools package? Don't think they are really needed for the library?

**Comment by Begasus** [comment]
For REQUIRES_devel, I don't see any mention for libjpeg, libpng, libtiff or libz when checking with pkg-config, are they really needed?

**Comment by Begasus** [comment]
This fixes:
```
~/haikuports> pkgman full-sync
100% repochecksum-1 [65 bytes]
Validating checksum for Haiku...done.
100% repochecksum-1 [64 bytes]
 Validating checksum for HaikuPorts...done.
Encountered problems:
problem 1: package lcms_devel-2.9-3 requires devel:libpng>=15.30.0, but none of the providers can be installed
solution 1:
- allow deinstallation of libpng16_devel-1.6.37-1
- allow deinstallation of freetype_devel-2.9.1-1
- allow deinstallation of poppler066_devel-0.66.0-2
 - allow deinstallation of wxqt_devel-3.1.3-1
- allow deinstallation of libxfont_devel-1.5.4-3
- allow deinstallation of fontconfig_devel-2.13.92-1
- allow deinstallation of libxft_devel-2.3.3-1
- allow deinstallation of cairo_devel-1.16.0-1
solution 2:
 - keep lcms_devel-2.9-4 from excluded repository
Please select a solution, skip the problem for now or quit.
```

File: media-libs/lcms/lcms-2.9.recipe (hunk 1 above)
Comment (korli) [reply media-libs/lcms/lcms-2.9.recipe]: not required, could be another PR as you wish.

**Comment by korli** [comment]
> For REQUIRES_devel, I don't see any mention for libjpeg, libpng, libtiff or libz when checking with pkg-config, are they really needed?

The headers could reference headers from these libs. Maybe check there.

**Comment by korli** [comment]
I see they were added by fbrosson in 4499968f08e5725f1343fe0745929fb93264299e

File: media-libs/lcms/lcms-2.9.recipe (hunk 1 above)
Comment (Begasus) [reply media-libs/lcms/lcms-2.9.recipe]: Or I could move them in this PR

**Comment by Begasus** [comment]
This fixes:
```
~/haikuports> pkgman full-sync
100% repochecksum-1 [65 bytes]
Validating checksum for Haiku...done.
100% repochecksum-1 [64 bytes]
 Validating checksum for HaikuPorts...done.
Encountered problems:
problem 1: package lcms_devel-2.9-3 requires devel:libpng>=15.30.0, but none of the providers can be installed
solution 1:
- allow deinstallation of libpng16_devel-1.6.37-1
- allow deinstallation of freetype_devel-2.9.1-1
- allow deinstallation of poppler066_devel-0.66.0-2
 - allow deinstallation of wxqt_devel-3.1.3-1
- allow deinstallation of libxfont_devel-1.5.4-3
- allow deinstallation of fontconfig_devel-2.13.92-1
- allow deinstallation of libxft_devel-2.3.3-1
- allow deinstallation of cairo_devel-1.16.0-1
solution 2:
 - keep lcms_devel-2.9-4 from excluded repository
Please select a solution, skip the problem for now or quit.
```

> I see they were added by fbrosson in [4499968](https://github.com/haikuports/haikuports/commit/4499968f08e5725f1343fe0745929fb93264299e)

OK, I'll leave them in then

**Comment by Begasus** [comment]
> > For REQUIRES_devel, I don't see any mention for libjpeg, libpng, libtiff or libz when checking with pkg-config, are they really needed?
> 
> The headers could reference headers from these libs. Maybe check there.

Seems like the "libs" are needed for the cmd's (not for the main nor devel package)
I doubt that even libpng16 is even needed in the recipe (checking here)

**Comment by korli** [comment]
then remove devel:libpng

File: media-libs/lcms/lcms-2.9.recipe [hunk 2]
```diff
@@ -68,6 +61,25 @@ BUILD_PREREQUIRES="
...
+	"
+REQUIRES_tools="
+	lcms$secondaryArchSuffix == $portVersion base
+	haiku$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix >= 9
```
Comment (korli) [inline media-libs/lcms/lcms-2.9.recipe]: >= 9 is not needed.

Comment (Begasus) [reply media-libs/lcms/lcms-2.9.recipe]: Correct only needed for devel

File: media-libs/lcms/lcms-2.9.recipe [hunk 3]
```diff
@@ -68,6 +61,25 @@ BUILD_PREREQUIRES="
 	cmd:make
 	"
 
+PROVIDES_tools="
+	lcms${secondaryArchSuffix}_tools = $portVersion
+	cmd:jpgicc$commandSuffix = 3.2
+	cmd:linkicc$commandSuffix = 2.2
+	cmd:psicc$commandSuffix = 2.1
+	cmd:tificc$commandSuffix = 6.2
+	cmd:transicc$commandSuffix = 4.3
```
Comment (korli) [inline media-libs/lcms/lcms-2.9.recipe]: what are these version numbers?

Comment (Begasus) [reply media-libs/lcms/lcms-2.9.recipe]: running the cmd's in Terminal with "-v" yields their version

Comment (Begasus) [reply media-libs/lcms/lcms-2.9.recipe]: searching github they aren't used by any other recipe

Comment (korli) [reply media-libs/lcms/lcms-2.9.recipe]: I don't really see the benefits of having these versions.

Comment (Begasus) [reply media-libs/lcms/lcms-2.9.recipe]: I'll remove them then :)

**Pull request merged**